src/native/libs/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -235,7 +235,7 @@ void InitializeOpenSSLShim(void)
 #if defined(TARGET_ARM) && defined(TARGET_LINUX)
     // This value will represent a time in year 2038 if 64-bit time is used,
     // or 1901 if the lower 32 bits are interpreted as a 32-bit time_t value.
-    time_t timeVal = (time_t)INT_MAX + 1;
+    time_t timeVal = (time_t)((unsigned long)INT_MAX + 1u);
```

Comment:
Why "unsigned long" instead of, e.g. uint32_t?

Reply:
Now that I think about it, both are kind of wrong. This should just be `unsigned`.

Reply:
Even more thinking about it, I don't know why we were doing `INT_MAX + 1`. We could just use `0x80000000` and convert it to a time_t. If `time_t` is 64-bit, it will be 2038. If it is 32-bit, it will be -2147483648, which is 1901. The conversion is enough to detect overflows.